lib/entities/bulk-action.ts

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import copy from 'fast-copy'
+import {
+  DefaultElements,
+  ISO8601Timestamp,
+  Link,
+  MakeRequest,
+  VersionedLink,
+} from '../common-types'
+import enhanceWithMethods from '../enhance-with-methods'
+import { BulkActionProcessingOptions, pollBulkActionStatus } from '../methods/bulk-action'
+
+/** Entity types supported by the BulkAction API */
+type Entity = 'Entry' | 'Asset'
+
+export type BulkActionType = 'publish' | 'unpublish' | 'validate'
+
+/** Represents the state of the BulkAction */
+export enum BulkActionStatus {
+  /** BulkAction is pending execution */
+  created = 'created',
+  /** BulkAction has been started and pending completion */
+  inProgress = 'inProgress',
+  /** BulkAction was completed successfully (terminal state) */
+  succeeded = 'succeeded',
+  /** BulkAction failed to complete (terminal state) */
+  failed = 'failed',
+}
+
+const STATUSES = Object.values(BulkActionStatus)
+type BulkActionStatuses = typeof STATUSES[number]
+
+interface BulkActionFailedError {
+  sys: { type: 'Error'; id: 'BulkActionFailed' }
+  details: {
+    errors: any
+    [key: string]: any
```

Comment:
Do we need the `[key: string] :any` here? We know the shape and I don't think we need to allow other parties to define additional fields. Also we know the shape of the errors

Reply:
*boops*